app/clients/letter/dvla.py

```diff
@@ -87,44 +84,73 @@ def name(self):
 
     @property
     def jwt_token(self):
-        if not self._jwt_token or time.time() >= self._jwt_expires_at:
+        # if the jwt is about to expire, just reset it ourselves to avoid unnecessary 401s
+        buffer = 60
+        if not self._jwt_token or time.time() + buffer >= self._jwt_expires_at:
             self._jwt_token = self.authenticate()
             jwt_dict = jwt.decode(self._jwt_token, options={"verify_signature": False})
             self._jwt_expires_at = jwt_dict["exp"]
 
         return self._jwt_token
 
+    def _handle_common_dvla_errors(self, e: requests.HTTPError):
+        if e.response.status_code == 429:
+            raise DvlaThrottlingException() from e
+        elif e.response.status_code >= 500:
+            raise DvlaRetryableException() from e
+        else:
+            raise DvlaNonRetryableException() from e
+
     def authenticate(self):
         """
         Fetch a JWT from the DVLA API that can be used in other DVLA API requests
         """
-        response = self.request.post(
-            f"{current_app.config['DVLA_API_BASE_URL']}/thirdparty-access/v1/authenticate",
-            json={
-                "userName": self.dvla_username.get(),
-                "password": self.dvla_password.get(),
-            },
-        )
-        response.raise_for_status()
-
-        return response.json()["id-token"]
-
-    def change_api_key(self):
-        from app import redis_store
-
-        with redis_store.get_lock(f"dvla-change-api-key-{self.dvla_username.get()}", timeout=60, blocking=False):
+        try:
             response = self.request.post(
-                f"{current_app.config['DVLA_API_BASE_URL']}/thirdparty-access/v1/new-api-key",
-                headers={
-                    "x-api-key": self.dvla_api_key.get(),
-                    "Authorization": self.jwt_token,
-                },
+                f"{current_app.config['DVLA_API_BASE_URL']}/thirdparty-access/v1/authenticate",
                 json={
                     "userName": self.dvla_username.get(),
                     "password": self.dvla_password.get(),
                 },
             )
             response.raise_for_status()
+        except requests.HTTPError as e:
+            if e.response.status_code == 401:
+                # likely the old password has already expired
+                current_app.logger.exception("Failed to generate a DVLA jwt token")
```

Comment:
Do we want to log an exception here if we are going to retry the task? I don't know if this will trigger an alert/email but I don't think we want that if it's going to automatically recover.

Reply:
good point - i think this will trigger an email if it happens in two consecutive minutes, but not otherwise (including if it errors multiple times in one minute) - as per https://github.com/alphagov/notifications-aws/pull/1141

Reply:
For now - with these email alerting rules.

If (when 🤞) we start using something like Sentry, this exception line and the raised error will be considered different errors/problems and alert separately. We can merge them but I think it seems like a not-very-useful error message to get when the exception below will either 1) retry automatically or 2) be the last retry and error out itself